examples/svi_horovod.py

```diff
@@ -0,0 +1,159 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+# Distributed training via Horovod.
+#
+# This tutorial demonstrates how to distribute SVI training across multiple
+# machines (or multiple GPUs on one or more machines) using the Horovod
+# library. Horovod enables data-parallel training by aggregating stochastic
+# gradients at each step of training. Horovod is not intended for model
+# parallelism. We focus on integration between Horovod and Pyro. For further
+# details on distributed computing with Horovod, see
+#   https://horovod.readthedocs.io/en/stable
+#
+# This assumes you have installed horovod, e.g. via
+#   pip install pyro[horovod]
+# For detailed instructions see
+#   https://horovod.readthedocs.io/en/stable/install.html
+# On my mac laptop I was able to install horovod with
+#   CFLAGS=-mmacosx-version-min=10.15 HOROVOD_WITH_PYTORCH=1 \
+#   pip install --no-cache-dir 'horovod[pytorch]'
+#
+# Finally, you'll need to run this script via horovodrun, e.g.
+#   horovodrun -np 2 python svi_horovod.py
+# For details on running Horovod see
+#   https://github.com/horovod/horovod/blob/master/docs/running.rst
+
+import argparse
+
+import torch
+import torch.multiprocessing as mp
+
+import pyro
+import pyro.distributions as dist
+from pyro.infer import SVI, Trace_ELBO
+from pyro.infer.autoguide import AutoNormal
+from pyro.nn import PyroModule
+from pyro.optim import Adam, HorovodOptimizer
+
+
+# We define a model as usual, with no reference to Horovod. This model is data
+# parallel and supports subsampling.
+class Model(PyroModule):
+    def __init__(self, size):
+        super().__init__()
+        self.size = size
+
+    def forward(self, covariates, data=None):
+        coeff = pyro.sample("coeff", dist.Normal(0, 1))
+        bias = pyro.sample("bias", dist.Normal(0, 1))
+        scale = pyro.sample("scale", dist.LogNormal(0, 1))
+        with pyro.plate("data", self.size, len(covariates)):
```

Comment:
so what happens if the user relies on `plate`-controlled subsampling? the same computation happens on all workers?

Reply:
Good point, I've added a couple sentences explaining requirements.